test/filtering/utilities.py

```diff
@@ -0,0 +1,78 @@
+import json
+import numpy as np
+import xarray as xr
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from harmony import Client
+
+
+def print_success(msg: str) -> None:
+    print(f'\033[92mSuccess: {msg}\033[0m')
+
+
+def print_error(msg: str) -> None:
+    print(f'\033[91mError: {msg}\033[0m')
+
+
+def validate_filter_outputs(harmony_client: Client, harmony_job_id: str) -> None:
+    """
+    Download the one output file, look up its truth entry in filter_truth.json
+    (which now carries "group/variable"), open that group, confirm the var exists,
+    count non‐NaNs, and assert it matches the expected count.
+    """
+    # 1) load the truth table
+    truth_path = Path(__file__).parent / "reference_data" / "filter_truth.json"
+    with open(truth_path, 'r') as f:
+        truth = json.load(f)
+
+    # 2) download
+    with TemporaryDirectory() as tmp_dir:
+        outs = [
+            future.result()
+            for future in harmony_client.download_all(
+                harmony_job_id, overwrite=True, directory=tmp_dir
+            )
+        ]
+        assert len(outs) == 1, f"Expected 1 output file, got {len(outs)}"
+        out_path = Path(outs[0])
+        fname = out_path.name
+        print_success(f"Got one filtered output file: {fname}")
+
+        # strip numeric prefix
+        suffix = fname.split('_', 1)[1]
+        assert suffix in truth, f"No truth entry for '{suffix}'"
+        entry = truth[suffix]
+
+        # 3) parse "group/variable"
+        group_var = entry["variable"]
+        if "/" in group_var:
+            group, var = group_var.split("/", 1)
```

Comment:
This looks like it should work if there is there is a single-nested variable (e.g., `group/variable`), but it won't work if you have more nested variables (e.g., `group/nested_group/variable`).

An alternative might be using [str.rpartition](https://docs.python.org/3/library/stdtypes.html#str.rpartition) (thanks to Vu for using this recently in net2cog!):

```
group, _, var = group_var.rpartition('/')
```

In the case of a non-nested variable, the `group` will be an empty string.

(I'm guessing this is relying on the TEMPO data not having more nested group structure, but I'm not sure if the regression tests will always only be using TEMPO for reference data)